WeatherToday/Sources/Data/NetworkAPI/Core/RequestConfiguration.swift

```diff
@@ -2,38 +2,55 @@
 //  RequestConfiguration.swift
 //
 
-import Alamofire
 import Foundation
+import Moya
 
-protocol RequestConfiguration {
-
-    var baseURL: String { get }
-
-    var endpoint: String { get }
-
-    var method: HTTPMethod { get }
-
-    var url: URLConvertible { get }
-
-    var parameters: Parameters? { get }
+enum RequestConfiguration {
+    case weather(cityName: String)
+}
 
-    var encoding: ParameterEncoding { get }
+extension RequestConfiguration: TargetType {
```

Comment:
Is this the on API? If we will have more endpoints I suggest creating a new file for each endpoint. `TargetType+Weather.swift`

Reply:
@blyscuit 
- After creating new file should I rename the `enum RequestConfiguration` to `enum WeatherRequestConfiguration`?
- By `I suggest creating a new file for each endpoint` did you mean creating new file for each `Target`? like one `Target` can contain multiple cases with different paths/endpoints like [here](https://www.kodeco.com/5121-moya-tutorial-for-ios-getting-started) but one `Target` is for one `baseUrl` ?

Reply:
- I would say that it's easier to declare a new enum for each sub-service (similar APIs) because how `enum` in Swift works.
- One Target enum can have multiple endpoints, divided by the business use case OR by the endpoints.

If we will only have one/two endpoints in this project then the refactor is not needed.